src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4584,21 +4623,26 @@ private SeparatedSyntaxList<ParameterSyntax> ParseParameterList(
             out SyntaxToken open,
             out SyntaxToken close,
             SyntaxKind openKind,
-            SyntaxKind closeKind)
+            SyntaxKind closeKind,
+            bool forExtension)
         {
             open = this.EatToken(openKind);
 
             var saveTerm = _termState;
             _termState |= TerminatorState.IsEndOfParameterList;
 
+            Func<LanguageParser, ParameterSyntax> parseElement = forExtension
+                    ? static @this => @this.ParseParameter(allowOptionalIdentifier: true)
+                    : static @this => @this.ParseParameter(allowOptionalIdentifier: false);
+
             var parameters = ParseCommaSeparatedSyntaxList(
                 ref open,
                 closeKind,
                 static @this => @this.IsPossibleParameter(),
-                static @this => @this.ParseParameter(),
+                parseElement,
                 skipBadParameterListTokens,
                 allowTrailingSeparator: false,
-                requireOneElement: false,
+                requireOneElement: forExtension, // For extension declarations, we require one receiver parameter
```

Comment:
>requireOneElement: forExtension [](http://example.com/codeflow?start=16&length=31)

It feels like it might be good to suppress any parsing diagnostics from syntax errors in the second and any following parameter in the list. Not sure how easy to do this, given that diagnostics is attached to nodes. Consider making a note to follow up later.